### PR TITLE
Added RecoveryServices backup schemas

### PR DESCRIPTION
**Pull request by DheerendraRathor**
Updated RecoveryServices schemas for Backup Services based on PR https://github.com/Azure/azure-rest-api-specs/pull/2208

### CODE REVIEW

File: schemas/2016-12-01/vaults/backupFabrics/protectionContainers.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# Microsoft.RecoveryServices/vaults/backupFabrics/protectionContainers template reference
```
Comment (Tiano2017) [inline schemas/2016-12-01/vaults/backupFabrics/protectionContainers.md]: md files are now allowed in this repo. please remove them.

**Comment by Tiano2017** [comment]
please add test for your new schema.

**Comment by DheerendraRathor** [comment]
@Tiano2017 Added tests and removed md files.

**Comment by Tiano2017** [comment]
@DheerendraRathor there are test cases failing. I also see "$ref '/definitions/Microsoft.Sql/servers/databases' in 'https://schema.management.azure.com/schemas/2016-12-01/Microsoft.RecoveryServices.json' can not be resolved." in my VS validation.

File: schemas/2016-12-01/Microsoft.RecoveryServices.json [hunk 2]
```diff
@@ -0,0 +1,3033 @@
...
+        },
+        "oldestRecoveryPoint": {
+          "type": "string",
+          "format": "date-time",
+          "description": "The oldest backup copy available."
+        },
+        "recoveryPointCount": {
+          "oneOf": [
+            {
+
```
Comment (DheerendraRathor) [inline schemas/2016-12-01/Microsoft.RecoveryServices.json]: @Tiano2017 Here is the '/definitions/Microsoft.Sql/servers/databases'.
Test tools are unable to recognize these kind of defintions where `/` is embedded in name itself. 

This is the reason tests are also failing.

Comment (Tiano2017) [reply schemas/2016-12-01/Microsoft.RecoveryServices.json]: That's an issue, right? I was using VS code validation tools to do this, and it shows errors. In that case we can't publish it, right?

Comment (DheerendraRathor) [reply schemas/2016-12-01/Microsoft.RecoveryServices.json]: I'll say that the validation tools need to be updated if they can't handle these scenario which are generated from perfectly valid swagger spec (from where SDKs have been shipped). 
So this is not an issue in definitions, rather a bug in tools.

Comment (DheerendraRathor) [reply schemas/2016-12-01/Microsoft.RecoveryServices.json]: @Tiano2017 What's the update on this? Are validation tools are going to get updated?

Comment (Tiano2017) [reply schemas/2016-12-01/Microsoft.RecoveryServices.json]: @DheerendraRathor We can't update the validation tool. We use the same validation library that VS Code is using. If this is rejected by our tools, it's going to be rejected by VS Code as well. I would recommend you update your swagger, or you'll need to ask VS Code to support this format.

**Comment by DheerendraRathor** [comment]
@Tiano2017 Can you please have a look at the updated PR?

**Comment by Tiano2017** [comment]
@DheerendraRathor please sync from master and resolve the conflicts.

**Comment by DheerendraRathor** [comment]
@Tiano2017 I've updated pull request and resolved conflicts with master

File: schemas/2015-01-01/deploymentTemplate.json [hunk 3]
```diff
@@ -555,6 +555,10 @@
                   { "$ref": "https://schema.management.azure.com/schemas/2018-03-01/Microsoft.BatchAI.json#/resourceDefinitions/clusters" },
                   { "$ref": "https://schema.management.azure.com/schemas/2018-03-01/Microsoft.BatchAI.json#/resourceDefinitions/fileServers" },
                   { "$ref": "https://schema.management.azure.com/schemas/2018-03-01/Microsoft.BatchAI.json#/resourceDefinitions/jobs" }
```
Comment (TianoMS) [inline schemas/2015-01-01/deploymentTemplate.json]: >} [](start = 134, length = 1)

missing a comma at the end of line.

**Comment by DheerendraRathor** [comment]
@Tiano2017 @TianoMS Updated PR.

**Pull request merged**